### PR TITLE
[SPARK-27155][TEST] Parameterize Oracle docker image name

### PR DESCRIPTION
**Pull request by zhulipeng**
## What changes were proposed in this pull request?

Update Oracle docker image name.

## How was this patch tested?

./build/mvn test -Pdocker-integration-tests -pl :spark-docker-integration-tests_2.12

### CODE REVIEW

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala [hunk 1]
```diff
@@ -55,7 +56,7 @@ class OracleIntegrationSuite extends DockerJDBCIntegrationSuite with SharedSQLCo
   import testImplicits._
 
   override val db = new DatabaseOnDocker {
-    override val imageName = "wnameless/oracle-xe-11g:16.04"
+    override val imageName = "deepdiver/docker-oracle-xe-11g:2.0"
```
Comment (wangyum) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: Why do we need update this image?

Comment (felixcheung) [reply external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: hmm.. who owns/licenses these images...

Comment (lipzhu) [reply external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: > Why do we need update this image?

image

Comment (srowen) [reply external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: I see, but is the new image 'legal' either? is there an official one? and if not, should we be using or testing against it?

Comment (dongjoon-hyun) [reply external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: I also have the same concerns on the legal stuff. If we don't have an official docker image, we had better remove this from docker test.

Comment (lipzhu) [reply external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: Seems there is no official Oracle docker image,  can we Ignore this Test to pass the docker integration test?

Comment (srowen) [reply external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: I think it should be deleted if there is no legitimate way to test this

Comment (felixcheung) [reply external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: agreed there..

Comment (srowen) [reply external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: Looks like this is how you build it:
https://github.com/oracle/docker-images/blob/master/OracleDatabase/SingleInstance/README.md
but looks like indeed you have to supply the binaries as they're not freely available.
Yes at best I think we'd need to disable this test and parameterize it, as it's not going to work reliably anyway.

**Comment by lipzhu** [comment]
Add a parameter SPARK_ORACLE_DOCKER_IMAGE_NAME for Oracle tests.

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala [hunk 2]
```diff
@@ -55,7 +55,8 @@ class OracleIntegrationSuite extends DockerJDBCIntegrationSuite with SharedSQLCo
   import testImplicits._
 
   override val db = new DatabaseOnDocker {
-    override val imageName = "wnameless/oracle-xe-11g:16.04"
+    override val imageName = sys.env.getOrElse("SPARK_ORACLE_DOCKER_IMAGE_NAME",
+      "wnameless/oracle-xe-11g:16.04")
```
Comment (srowen) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: We can't even have a default; this default doesn't work

**Review by srowen — CHANGES_REQUESTED** [review]
Is there a way to disable this test also, if it's run automatically? or at least skip its execution if no image is specified?

**Comment by srowen** [comment]
OK, is the point that this will never be run anyway unless someone runs it manually? Then I think this is OK if you remove the default (that doesn't exist anyway) and add a few more comments pointing to the oracle repo about how the image can be built.

**Review by srowen — APPROVED** [review]

**Comment by dilipbiswal** [comment]
Should we update the PR description to reflect what its doing now in case some one refers to it later ?

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala [hunk 3]
```diff
@@ -39,13 +39,16 @@ import org.apache.spark.tags.DockerTest
  * while Spark QA test run.
  *
  * The following would be the steps to test this
- * 1. Pull oracle 11g image - docker pull wnameless/oracle-xe-11g
- * 2. Start docker - sudo service docker start
- * 3. Download oracle 11g driver jar and put it in maven local repo:
+ * 1. Build Oracle database in Docker, please refer below link about how to.
+ *    @see https://github.com/oracle/docker-images/blob/master/OracleDatabase/SingleInstance/README.md
```
Comment (srowen) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: The line is too long, and the `@see` tag doesn't work here I think. Put "See" on the previous line and unindent to see if that gets it to under 100 characters

**Comment by lipzhu** [comment]
reset this please.

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala [hunk 4]
```diff
@@ -78,11 +81,15 @@ class OracleIntegrationSuite extends DockerJDBCIntegrationSuite with SharedSQLCo
     conn.prepareStatement(
       "CREATE TABLE ts_with_timezone (id NUMBER(10), t TIMESTAMP WITH TIME ZONE)").executeUpdate()
     conn.prepareStatement(
-      "INSERT INTO ts_with_timezone VALUES " +
-        "(1, to_timestamp_tz('1999-12-01 11:00:00 UTC','YYYY-MM-DD HH:MI:SS TZR'))").executeUpdate()
+      """
+        |INSERT INTO ts_with_timezone VALUES (
+        |1, to_timestamp_tz('1999-12-01 11:00:00 UTC','YYYY-MM-DD HH:MI:SS TZR'))
+         """.stripMargin).executeUpdate()
```
Comment (dongjoon-hyun) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: indentation?

Comment (srowen) [reply external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: Or @lipzhu why did this part change? I may be missing it but it seems like just a formatting change. I don't think it's necessary

Comment (lipzhu) [reply external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: @srowen Yes, it is just a formatting change, I see these 2 codes' style is not consistent with the others in this suite. If you think it it not necessary, I will change them back.

**Review by srowen — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by dilipbiswal** [comment]
@dongjoon-hyun Hi, i was trying to follow the instructions in OracleIntegrationSuite to try and build an image to run the oracle tests. I was trying it on mac.. Haven't been able to. Were you able to run this successfully ? Please let me know .. I get the following error when i try to build the image. 

```
Building image 'oracle/database:11.2.0.2-xe' ...
Sending build context to Docker daemon  2.351GB
Step 1/10 : FROM oraclelinux:7-slim
 ---> f7512ac13c1b
Step 2/10 : MAINTAINER Gerald Venzl <gerald.venzl@oracle.com>
 ---> Running in 626a011e37a1
Removing intermediate container 626a011e37a1
 ---> 3a1f198c974f
Step 3/10 : ENV ORACLE_BASE=/u01/app/oracle     ORACLE_HOME=/u01/app/oracle/product/11.2.0/xe     ORACLE_SID=XE     INSTALL_FILE_1="oracle-xe-11.2.0-1.0.x86_64.rpm.zip"     INSTALL_DIR="$HOME/install"     CONFIG_RSP="xe.rsp"     RUN_FILE="runOracle.sh"     PWD_FILE="setPassword.sh"     CHECK_DB_FILE="checkDBStatus.sh"
 ---> Running in 969673e55df7
Removing intermediate container 969673e55df7
 ---> 441b63aabd8a
Step 4/10 : ENV PATH=$ORACLE_HOME/bin:$PATH
 ---> Running in 52469448bf1b
Removing intermediate container 52469448bf1b
 ---> b80dbfddd0a9
Step 5/10 : COPY $INSTALL_FILE_1 $CONFIG_RSP $RUN_FILE $PWD_FILE $CHECK_DB_FILE $INSTALL_DIR/
COPY failed: stat /var/lib/docker/tmp/docker-builder372031542/oracle-xe-11.2.0-1.0.x86_64.rpm.zip: no such file or directory
```

**Comment by dongjoon-hyun** [comment]
Hi, @dilipbiswal .
This PR doesn't imply that `oracle/database:11.2.0.2-xe` works, does this?

**Comment by srowen** [comment]
Yeah, I'm not clear one would be able to use this at all without an Oracle license. I guess I'd be surprised if there's any free public Oracle DB image out there.

**Comment by dilipbiswal** [comment]
@dongjoon-hyun Oh... actually i just needed a little help to get the oracle tests to run as i am trying to fix a defect that would require me to add a test in this suite. I just used this recent PR to ask the question :-)

**Comment by dilipbiswal** [comment]
@dongjoon-hyun A quick update: I got hold of an un-official image and got the tests to pass. So i am not blocked at the moment :-)

**Pull request closed without merging**